src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -1743,7 +1743,7 @@ bool DoorLockServer::findUserIndexByCredential(chip::EndpointId endpointId, Cred
             if (credentialInfo.credentialData.data_equal(credentialData))
             {
                 userIndex       = i;
-                credentialIndex = i;
+                credentialIndex = credential.CredentialIndex;
```

Comment:
Nice catch, but can we please add a YAML test for this?

Reply:
YAML test has been updated.

Reply:
No, it hasn't.  What was updated was:

1. One of the certification tests.  Those should not get updated without corresponding test plan changes.
2. Not one of the tests that actually runs anything automated.  What should have been done is either updating src/app/tests/suites/DL_UsersAndCredentials.yaml or adding a new test.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/24992 to track this....